### DISCUSSION TITLE
Prefer TypeScript?

### DISCUSSION THREAD

**Discussion by septs** [OpenINF/.github#24] [Ideas]
I think strongly typed code helps maintain.

**Comment by DerekNonGeneric** [comment]
Yes, absolutely. I am setting up each project individually right now to see what each one needs as far as @typescript-eslint is concerned then I plan on putting that in a shareable config so that we don't really have to think about it anymore w/ [`@openinf/eslint-config`](https://github.com/openinf/inf-toolset/tree/main/workspaces/eslint-config) we can just install it and extend the preset. I also like to ship typings, but so far feel that tests might need to be written in JS to _actually_ know what is going on behind the scenes w/o transpilers fooling around.

**Comment by septs** [comment]
suggest, create `@openinf/eslint-plugin`

like https://github.com/DimensionDev/eslint-plugin

pre-ruleset and customize rule set in one package

documentation: see https://eslint.org/docs/latest/developer-guide/working-with-plugins

**Comment by DerekNonGeneric** [comment]
> You can also use scoped packages in the format of @<scope>/eslint-plugin-<plugin-name> such as @jquery/eslint-plugin-jquery or even @<scope>/eslint-plugin such as @jquery/eslint-plugin.
> &mdash; https://eslint.org/docs/latest/developer-guide/working-with-plugins

Do you think we should use [`@openinf/eslint-plugin`](https://github.com/openinf/inf-toolset/tree/main/workspaces/eslint-plugin) for the name here or should we specify something else like `@openinf/eslint-plugin-prettier` to get that handled for all prettier projects?

**Reply by DerekNonGeneric** [reply]
As another example, we could also maybe have a layered approach to this w/ `@openinf/eslint-plugin` just specifying the parser we are intending to use (would be `@typescript-eslint/parser` I guess), but maybe that should be `@openinf/eslint-plugin-typescript` so that we can always fall back to using the babel parser if necessary.

**Reply by septs** [reply]
I think this problem, need split to new dicussion (package naming style design?)

**Reply by septs** [reply]
https://docs.npmjs.com/cli/v8/commands/npm-init#description

<img width="639" alt="CleanShot 2022-08-24 at 19 13 04@2x" src="https://user-images.githubusercontent.com/3842474/186404717-df033cf0-181a-406d-99c4-4527b0186352.png">

etc...

**Reply by DerekNonGeneric** [reply]
> I think this problem, need split to new dicussion (package naming style design?)

Ah yeah, but not sure they are all consistent. I have been trying to make an effort to document all of the correct naming conventions in the https://github.com/openinf/inf-toolset/tree/main/workspaces directory, but maybe it would be more worthwhile to link the docs on the README or commit message to be sure about it.

I was not aware that `npm init` cares about `create` prefix. I thought that was just for project generators like Yeoman and stuff. Can you please start a new discussion about this? I probably have blind spots because I never create projects using generators that need to be named with `@scope/create-` prefixes like that.

**Reply by DerekNonGeneric** [reply]
Oh now I do remember reading about customizing the `npm init` command&hellip; Maybe this discussion should be had in the https://github.com/openinf/inf-toolset repo instead? Anyways, I just invited you w/ `maintain` privileges there, so if you feel it would be better scoped in that repo, feel free.

**Reply by DerekNonGeneric** [reply]
The reason why that repo is named `inf-toolset` instead of just `toolset` is because I was planning to imitate the https://github.com/github/rubocop-github style of partially revealing closed-source via config:

> This repository provides recommended RuboCop configuration and additional Cops for use on GitHub open source and internal Ruby projects.

I am certainly open to changing the name. In fact, I had already thought that if that sort of mentality might be an issue, we can put all these configs in another repo like https://github.com/openinf/openinf (the naming can also be changed, but check the description of the project).

**Reply by septs** [reply]
I think, this org need a specificion repo, save {package,repo} naming style specificion, etc other specification or standard.

**Reply by DerekNonGeneric** [reply]
Yeah, a place to document coding conventions/style guide is a pretty good idea. Much of what https://github.com/openinf/inf-toolset was supposed to do w/ lint rules was documenting the preferences, but not immediately sure what to name the repo:
- `openinf/style-guides`
- `openinf/coding-conventions`
- `openinf/coding-standards`
- `openinf/specifications`

Is there an idiomatic repo name (like the community health files' names) that we should be using for this?

**Reply by septs** [reply]
I recommend, `openinf/standards`

to be extreme, e.g: protocol {specification,design}, file format {specification,design}, metadata file format (https://w3c.github.io/w3c.json.html), etc

**Reply by DerekNonGeneric** [reply]
Yeah, the org metadata file could also be specified in that repo. Hmm&hellip;
